omnigibson/tasks/task_base.py

```diff
@@ -193,6 +212,9 @@ def reset(self, env):
             termination_condition.reset(self, env)
         for reward_function in self._reward_functions.values():
             reward_function.reset(self, env)
+        for metric_function in self._metric_functions.values():
+            metric_function.reset(self, env)
+            metric_function._reward = 0.0
```

Comment:
Instead of putting this here, make sure *every* metric function sets reward=0 when reset. Then remove this